internal/pkg/sync/config.go

```diff
@@ -391,6 +397,41 @@ func (l *location) refreshAuth() error {
 	return fmt.Errorf("no authorization data for")
 }
 
+//
+func (l *location) refreshAuthGCP() error {
+	if l.expiry.Sub(time.Now()) > 0 {
+		return nil
+	}
+
+	var runner func() (string, time.Time, error)
```

Comment:
Just declaring `var authToken string, expiry time.Time, err error`, and then assigning values by calling appropriate function in corresponding `if` branch would be more straightforward.